backend/firebaseFunctions.js

```diff
@@ -269,3 +272,62 @@ export async function checkIfUserIsInCircle(circle) {
 	}
 	return inCircle;
 }
+//NRA
+export async function makeCircleUserList(circle) {
+	console.log("start makeCircleUserList");
+	console.log("Passed circle: " + circle);
+	let circleSettingsData = [];
+	let targetUserList = Object.keys(
+		(await readData(`prayer_circle/circles/${circle}/members/`)) || {}
+	);
+	console.log("Target user list " + targetUserList);
+
+	for (targetUser in targetUserList) {
+
+		let fname = Object.values(
+			(await readData(`prayer_circle/users/${targetUser}/public/fname`)) || {}
+		);
+		let lname = Object.values(
+			(await readData(`prayer_circle/users/${targetUser}/public/lname`)) || {}
+		);
+		let name = fname + " " + lname;
```

Comment:
```suggestion

		let data = await readData(`prayer_circle/users/${targetUser}/public`) || {}
		);
		let name = data.fname + " " + data.lname;
```
This is a more optimized way to access this information

Reply:
[Done]

Reply:
[done]